pkg/controller/clusterprovision/clusterprovision_controller_test.go

```diff
@@ -390,3 +439,62 @@ func getProvision(c client.Client) *hivev1.ClusterProvision {
 	}
 	return provision
 }
+
+type podOption func(*corev1.Pod)
+
+func testPod(nameSuffix string, opts ...podOption) *corev1.Pod {
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-%s", testJob().Name, nameSuffix),
+			Namespace: testNamespace,
+			Labels: map[string]string{
+				controllerUidLabelKey: testControllerUid,
+			},
+		},
+	}
+
+	for _, o := range opts {
+		o(pod)
+	}
+
+	return pod
+}
+
+func pending() podOption {
+	return func(pod *corev1.Pod) {
+		pod.Status.Phase = "Pending"
+	}
+}
+
+func running() podOption {
+	return func(pod *corev1.Pod) {
+		pod.Status.Phase = "Running"
+	}
+}
+
+func success() podOption {
+	return func(pod *corev1.Pod) {
+		pod.Status.Phase = "Succeeded"
+	}
+}
+
+func assertConditionStatus(t *testing.T, provision *hivev1.ClusterProvision, condType hivev1.ClusterProvisionConditionType, status corev1.ConditionStatus) {
+	found := false
+	for _, cond := range provision.Status.Conditions {
+		if cond.Type == condType {
+			found = true
+			assert.Equal(t, string(status), string(cond.Status), "condition found with unexpected status")
+		}
+	}
+	assert.True(t, found, "did not find expected condition type: %v", provision.Status.Conditions)
+}
+
+func assertConditionReason(t *testing.T, cd *hivev1.ClusterProvision, condType hivev1.ClusterProvisionConditionType, reason string) {
```

Comment:
Use the same logic in `assertConditionStatus` and `assertConditionReason`. Either return when the condition is found (my preference) or use a `found` variable.

Reply:
Done